pkg/vault/contracts/BatchRouter.sol

```diff
@@ -960,6 +979,18 @@ contract BatchRouter is IBatchRouter, BatchRouterStorage, RouterCommon, Reentran
                     userData: params.userData
                 })
             );
+
+            if (isStaticCall == false && kind == SwapKind.EXACT_OUT) {
+                // If SwapKind of wrap is EXACT_OUT, the limit of underlying tokens was taken from the user, so the
+                // difference between limit and exact underlying amount needs to be returned to the sender.
+                uint256 valueToSend;
+                unchecked {
+                    // `limits[i]` will always be bigger than `underlyingAmounts[i]` because
+                    // `erc4626BufferWrapOrUnwrap` checks it.
+                    valueToSend = limits[i] - underlyingAmounts[i];
+                }
```

Comment:
I think it's fine to use checked math here.

The comment is true, and even if it wasn't the transaction would revert. But i'd leave `unchecked` operations just for cases where it's blatantly obvious it won't over/underflow (i.e. `if (a > b) a - b`).

In this case the assumption is enforced in a separate contract, so I wouldn't say it's obvious just by looking at this code.

Reply:
It saves 30 bytes of bytecode, but since we're going to split the BatchRouter anyway, it makes sense. I'll leave it checked then.